src/services/productService.js

```diff
@@ -38,3 +38,11 @@ export function updateProduct(id, product) {
 export function deleteProduct(id) {
   return new Product({ id }).fetch().then((product) => product.destroy());
 }
+
+export async function stockProduct(id, product) {
+  const old = await getProduct(id).then((data) => data);
+
+  return new Product({ id }).save({
+    amount: Number(old.attributes.amount) + Number(product.amount)
+  });
```

Comment:
update to using `amount = amount + 1`.
https://stackoverflow.com/a/9294062/9002484